### ISSUE TITLE
[github automatic releases] use github's "Auto-generate release notes" feature

### ISSUE THREAD

**Issue by DetachHead** [marvinpinto/actions#312]
![image](https://user-images.githubusercontent.com/57028336/137098428-0232140a-8f9b-4799-be51-5f0ce5ad47d5.png)
![image](https://user-images.githubusercontent.com/57028336/137098702-d6b39b0d-31dc-4375-ad0b-c7e2de34b0fa.png)

**Comment by frankbolviken** [comment]
Yes please ;)

**Comment by marvinpinto** [comment]
Hey there! The GitHub release notes feature looks pretty cool! I am not sure how the `automatic-releases` action would help with it though?

If the GH release notes feature works for you, why would you still use the `automatic-releases` action?

**Comment by DetachHead** [comment]
From my understanding, it doesn't automate releases like this action does, it only generates the release notes. I'd imagine the `automatic-releases` action is still needed to actually create them

**Comment by marvinpinto** [comment]
@DetachHead Ahh, I gotcha. I'll consider it, and keep this open for now 👍🏾

**Comment by frankbolviken** [comment]
Yeah, you will have to manually go in and click "create release", then select tag etc, and click the "auto-generate release notes" and click publish. So would be nice if this action could also optionally execute this command somehow :)

**Comment by V3lop5** [comment]
I would also like to see this feature soon! 

Some additional information:
Docs can be found [here](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes).
For now I didn't find any official api for this.

For example when you edit the release with tag v0.0.1 and press the button a request to
https://github.com/:user:/:repo:/releases/notes?commitish=v0.0.1&tag_name=v0.0.1
is executed and returns a json containing the generated release notes as markdown.
I wonder if you could try query this endpoint.

**Comment by V3lop5** [comment]
@marvinpinto 
I think the REST API from GitHub now supports this feature:
https://docs.github.com/en/rest/reference/releases#create-a-release
see parameter `generate_release_notes`


> Whether to automatically generate the name and body for this release. If name is specified, the specified name will be used; otherwise, a name will be automatically generated. If body is specified, the body will be pre-pended to the automatically generated notes.
> Default: false



I'm not into js development, but should be quite easy to pass this parameter? 
Would be nice if somebody can pick this up - otherwise I'll try my luck in the next couple of weeks.

**Comment by peshrus** [comment]
https://docs.github.com/en/rest/reference/releases#generate-release-notes-content-for-a-release - This API can be used to generate the release notes.
@V3lop5 `generate_release_notes` parameter should also work.
@marvinpinto can I help with adding this?
